packages/clerk-js/src/ui/hooks/useInView.ts

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useRef, useState } from 'react';
+
+import type { IntersectionOptions } from '../components/OrganizationSwitcher/OtherOrganizationActions';
+
+export const useInView = (params: IntersectionOptions) => {
```

Comment:
It might be nice to drop leave a doc comment here explaining what this hook does (it might seem obvious, but it can be a helpful breadcrumb for the future)